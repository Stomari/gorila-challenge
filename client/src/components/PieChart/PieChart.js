import React from 'react';
import ReactChart from 'react-minimal-pie-chart';

const PieChart = ({ fixedIncome, variableIncome }) => {
  const fixedPercentage = parseFloat(((fixedIncome.length * 100) / (fixedIncome.length + variableIncome.length)).toFixed(2));
  const variablePercentage = parseFloat(((variableIncome.length * 100) / (fixedIncome.length + variableIncome.length)).toFixed(2));
  
  if (fixedPercentage || variablePercentage) {
    return (
      <section className="pie-chart">
        <div className="pie-info">
          <div className="row">
            <div className="fixed-legend"></div>
            <h4 className="fixed-percentage"><strong>{fixedPercentage}%</strong> da carteira em renda fixa</h4>
          </div>
          <div className="row">
            <div className="variable-legend"></div>
            <h4 className="variable-percentage"><strong>{variablePercentage}%</strong> da carteira em renda vari√°vel</h4>
          </div>
        </div>
        <ReactChart className="chart" data={[
          {
            title: 'fixed income',
            value: fixedPercentage,
            color: '#50c2c5'
          },
          {
            title: 'variable income',
            value: variablePercentage,
            color: '#6ade92'
          },
        ]}
        />
      </section>
    )
  } else {
    return (
      null
    )
  }
}

export default PieChart;