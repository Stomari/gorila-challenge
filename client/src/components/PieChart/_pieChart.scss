.pie-chart {
  @include displayLocal(center);
  align-items: center;
  padding: 30px 0 30px 0;

  .pie-info {
    padding-right: 50px;
    
    .fixed-legend {
      background-color: $bg-blue;
      width: 30px;
      height: 30px;
    }
    .variable-legend {
      background-color: $bg-green;
      width: 30px;
      height: 30px;
    }
    h4 {
      font-weight: normal;
      font-size: 1.4rem;
      margin-top: 5px;
      padding-left: 5px;
    }
  }

  .chart {
    width: 300px;
  }
}

@media (max-width: 1000px) {
  .pie-chart {
    flex-direction: column;
  }
}