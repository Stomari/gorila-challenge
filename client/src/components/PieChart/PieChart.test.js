import React from 'react';
import { shallow } from 'enzyme';
import PieChart from './PieChart';


describe('PieChart rendering without props', () => {
  const component = shallow(<PieChart fixedIncome={[]} variableIncome={[]} />);

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });
  
  it('should render form', () => {
    expect(component.type()).toEqual(null);
  });
});


describe('PieChart rendering with props', () => {
  const mockFixed = [{
    type: 'fixa',
    value: 'R$100,00',
    date: '2019-09-01'
  }];
  const mockVariable = [{
    type: 'variavel',
    value: 'R$200,00',
    date: '2019-09-02'
  }]
  const component = shallow(<PieChart fixedIncome={mockFixed} variableIncome={mockVariable} />);

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render .pie-chart', () => {
    expect(component.find('.pie-chart').length).toEqual(1);
  });

  it('should render .pie-info', () => {
    expect(component.find('.pie-info').length).toEqual(1);
  });

  it('should render correct fixed investments percentage', () => {
    expect(component.find('.fixed-percentage').text()).toEqual('50% da carteira em renda fixa');
  });

  it('should render correct variable investments percentage', () => {
    expect(component.find('.variable-percentage').text()).toEqual('50% da carteira em renda variÃ¡vel');
  });

  it('should render pie chart', () => {
    expect(component.find('.chart').length).toEqual(1);
  });

  it('pie chart should have data props', () => {
    expect(component.find('.chart').prop('data')).toEqual([
      {
        title: 'fixed income',
        value: 50,
        color: '#50c2c5'
      },
      {
        title: 'variable income',
        value: 50,
        color: '#6ade92'
      },
    ]);
  });
});