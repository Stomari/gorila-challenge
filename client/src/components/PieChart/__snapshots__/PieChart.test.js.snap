// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart rendering with props should render 1`] = `ShallowWrapper {}`;

exports[`PieChart rendering without props should render 1`] = `ShallowWrapper {}`;
